cartoframes/dataset.py

```diff
@@ -79,10 +79,11 @@ def from_geojson(cls, geojson):
     def upload(self, with_lnglat=None, if_exists=FAIL, table_name=None, schema=None, context=None):
         if table_name:
             self.table_name = normalize_name(table_name)
-        if schema:
-            self.schema = schema
         if context:
             self.cc = context
+            self.schema = context.get_default_schema()
```

Comment:
If the user has already a schema and he calls the `upload` function with a different one, the original is modified. Is this the expected behavior?

Reply:
Yes, I have said in the description setting the dataset context implies having a new schema

Reply:
True that

Reply:
I mean:
new context == new `api key` and new `username`
So if we have a new username (or the same), we have to update the schema.

Using a different schema is needed to use data from users from your organization and I can not see a situation of conflict with this approach

Reply:
I have only changed it here because it is the only place updating the context. In the `develop` we will need to do it in the `context` setter method